### PR TITLE
create out dir if it does not exist granting all permissions to the owner

### PR DESCRIPTION
**Pull request by giulianopz**
This PR fixes the issue of the program exiting in err due to the `/out` dir not being present. It creates such dir by granting to the owner all permissions.

### CODE REVIEW

File: internal/encrypt/encrypt.go [hunk 1]
```diff
@@ -88,13 +88,20 @@ func (e *Encrypter) Encrypt(reader io.Reader, filename string) error {
...
+const outDirName = "out"
+
 func (e *Encrypter) generateAndSaveMasterKey(filename string) ([]byte, error) {
 	masterKey, err := sss.GenerateMasterKey()
 	if err != nil {
 		return nil, err
 	}
 
-	err = file.WriteKey(masterKey, "out/"+filename)
+	if err := os.MkdirAll(outDirName, 0744); err != nil {
```
Comment (enrichman) [inline internal/encrypt/encrypt.go:99]: Is `0644` enough?

File: internal/encrypt/encrypt.go [hunk 2]
```diff
@@ -88,13 +88,20 @@ func (e *Encrypter) Encrypt(reader io.Reader, filename string) error {
...
+		fmt.Println(err)
```
Comment (enrichman) [inline internal/encrypt/encrypt.go:100]: I would drop the log, we are already returning the error and logging it before quitting (https://github.com/enrichman/stegosecrets/blob/main/cmd/stego/stego.go#L13)

**Review by enrichman — CHANGES_REQUESTED** [review]

File: internal/encrypt/encrypt.go (hunk 1 above)
Comment (giulianopz) [reply internal/encrypt/encrypt.go:99]: You need the 'execute' bit to traverse the dir and access its content, I think.

File: internal/encrypt/encrypt.go (hunk 2 above)
Comment (giulianopz) [reply internal/encrypt/encrypt.go:100]: Sorry, the log was just to test code. I will remove it.

File: internal/encrypt/encrypt.go (hunk 1 above)
Comment (enrichman) [reply internal/encrypt/encrypt.go:99]: That's correct, it works for a single dir but it fails for multiple subdirectories!

**Review by enrichman — APPROVED** [review]

**Pull request merged**